### PR TITLE
30 Frontend and Backend for Authentication

### LINKED ISSUE #30: Basic Login Authentication
## What needs to be done?
Being able to login as with email and password using the default Django JWT

## User Story
As a user, I can login using my email and password.

## Other information
This will be extended later on by #29

### PR DESCRIPTION
**Pull request by frinzekt**
## Change Summary
- initialise the API connection
- initialised the permission guards and context of the app
- initialised the API dj-rest-auth endpoints
- initialised the theme of the app
- initialised the login form with connection to the backend

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [x] The change has tests
- [x] The change has documentation

## Other Information
- This will not completely close #30 because the login form is too basic enough

### CODE REVIEW

**Comment by frinzekt** [comment]
The ones I have added documentations to are worth paying attention to

File: backend/authentication/tests/test_authentication.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    request = api_client.post(reverse("api-v1:rest_login"), {"username": "admin", "password": "Password123"})
+
+    # Check that the response is OK (JWT Created)
+    assert request.status_code == status.HTTP_200_OK
+    assert "access_token" in request.data
+    assert "refresh_token" in request.data
+    assert "user" in request.data
+
+    user = request.data["user"]
+    assert user["username"] == "admin"
```
Comment (frinzekt) [inline backend/authentication/tests/test_authentication.py:22]: test comment to demonstrate how to add comment

**Review by frinzekt — COMMENTED** [review]
general comment demo

**Comment by frinzekt** [comment]
@uwasystemhealth/redbacks ping... This PR has already been in review for 1 week. This ticket for logging in is a partial requirement for any frontend ticket that will need to be authenticated

**Review by Tommo303 — APPROVED** [review]
All good

**Pull request merged**